src/ProjectOrigin.Registry.Server/Services/RegistryService.cs

```diff
@@ -0,0 +1,70 @@
+using Grpc.Core;
+using MassTransit;
+using ProjectOrigin.Registry.V1;
+using ProjectOrigin.VerifiableEventStore.Services.EventStore;
+using ProjectOrigin.Registry.Server.Extensions;
+using System.Threading.Tasks;
+using System.Linq;
+using System;
+using ProjectOrigin.VerifiableEventStore.Services.TransactionStatusCache;
+
+namespace ProjectOrigin.Registry.Server;
+
+public class RegistryService : V1.RegistryService.RegistryServiceBase
+{
+    private IEventStore _eventStore;
+    private IBus _bus;
+    private ITransactionStatusService _transactionStatusService;
+
+    public RegistryService(IEventStore eventStore, IBus bus, ITransactionStatusService transactionStatusService)
+    {
+        _eventStore = eventStore;
+        _bus = bus;
+        _transactionStatusService = transactionStatusService;
+    }
+
+    public override async Task<SubmitTransactionResponse> SendTransactions(SendTransactionsRequest request, ServerCallContext context)
+    {
+        var jobs = request.Transactions.Select(transaction => TransactionJob.Create(transaction));
+        await _bus.PublishBatch(jobs);
+
+        request.Transactions.AsParallel().ForAll(async transaction =>
+        {
+            await _transactionStatusService.SetTransactionStatus(
+                transaction.GetTransactionId(),
+                new VerifiableEventStore.Models.TransactionStatusRecord(VerifiableEventStore.Models.TransactionStatus.Pending)
+                );
+        });
```

Comment:
Why running this in parallel? Why not just send all transactions in `request.Transaction` to the _transactionStatusService?